internal/compiler/generator/cpp.rs

```diff
@@ -2988,15 +3002,16 @@ fn compile_expression(expr: &llr::Expression, ctx: &EvaluationContext) -> String
         }
         Expression::ReadLocalVariable { name, .. } => ident(name).to_string(),
         Expression::StructFieldAccess { base, name } => match base.ty(ctx) {
-            Type::Struct { fields, name : None, .. } => {
-                let index = fields
-                    .keys()
-                    .position(|k| k == name)
-                    .expect("Expression::ObjectAccess: Cannot find a key in an object");
-                format!("std::get<{}>({})", index, compile_expression(base, ctx))
-            }
-            Type::Struct{..} => {
-                format!("{}.{}", compile_expression(base, ctx), ident(name))
+            Type::Struct(s)=> {
+                if let Some(name) = &s.name {
```

Comment:
this reverse the condition. (it used to check for None)
That's why the test fails.

Reply:
fixed, thanks for spotting!